Fix for 7086

rokm opened the pull request:
Remove the attempt at reading `top_level.txt` metadata from egg-installed packages. The obtained top-level path is not used anywhere, and the attempt at reading it triggers #7086 when `top_level.txt` is unavailable. Fixes #7086.

Update the installation instructions to discourage bad practices and reduce the amount of misinformation.

rokm committed:
building: remove spurious attempt to read distribution's top_level.txt

The code attempting to collect data from egg-installed packages
is trying to read distribution metadata's top_level.txt, but
never uses the result.

This spurious attempt at reading the file raise exception when a
package without top_level.txt happens to be installed as an egg
(typically due to being installed via the obsolete
`python setup.py install`).

rokm committed:
docs: update installation instructions

Remove references to `python setup.py install` and add a note
about its deprecation.

Add a note about invoking PyInstaller via `python -m PyInstaller`.

Remove the paragraph about running wrapper scripts from the source
directory, as those have been removed a while ago.

bwoodsend reviewed (approved):
_Execute ``python setup.py install`` with administrator privilege to install or upgrade PyInstaller._ :vomiting_face: Good riddance!

Are you not going to do [your bdist_egg override](https://github.com/pyinstaller/pyinstaller/issues/7086#issuecomment-1333642291)? I'm all for it.

rokm committed:
setup.cfg: use license_files instead of deprecated license_file

Silences a deprecation warning when building sdist using "build"
(or the deprecated `python setup.py`). Set minimum required version
of setuptools to 42.0.0, which introduced license_files.

rokm committed:
setup.py: prevent implicit building of python eggs

Prevent `python setup.py install` from implicitly building and
installing python eggs. Instead, display a message that installing
from source should be done via `pip install .`.

Based on jupyter/nbconvert@26f50e3a03426d547529be50b4bc86762f8510dd

rokm commented:
> _Execute `python setup.py install` with administrator privilege to install or upgrade PyInstaller._ 

Yeah, this is a prime example where fewer details would be better...

> Are you not going to do [your bdist_egg override](https://github.com/pyinstaller/pyinstaller/issues/7086#issuecomment-1333642291)? I'm all for it.

Done.
